js/core/option_manager.js

```diff
@@ -0,0 +1,215 @@
+            return;
+        }
+
+        this._changingCallbacks.fire(name, previousValue, value);
+
+        this._setValue(name, value, merge);
+        this._changedCallbacks.fire(name, value, previousValue);
+    }
+
+    _setRelevantNames(options, name, value) {
+        if(name) {
+            const alias = this._normalizeName(name);
+
+            if(alias && alias !== name) {
+                this._setField(options, alias, value);
+                this._clearField(options, name);
+            }
+        }
+    }
+
+    _normalizeName(name) {
+        if(name) {
+            let deprecate;
+            if(!this._cachedDeprecateNames.length) {
+                for(const optionName in this._deprecatedOptions) {
+                    this._cachedDeprecateNames.push(optionName);
+                }
+            }
+            for(let i = 0; i < this._cachedDeprecateNames.length; i++) {
+                if(this._cachedDeprecateNames[i] === name) {
+                    deprecate = this._deprecatedOptions[name];
+                    break;
+                }
+            }
+            if(deprecate) {
+                this._logWarningIfDeprecated(name);
+                const alias = deprecate.alias;
+
+                if(alias) {
+                    name = alias;
+                }
+            }
+        }
+
+        return name;
+    }
+
+    _prepareRelevantNames(options, name, value) {
+        if(typeUtils.isPlainObject(value)) {
+            for(const valueName in value) {
+                this._prepareRelevantNames(options, name + "." + valueName, value[valueName]);
+            }
+        }
+
+        this._setRelevantNames(options, name, value);
+    }
+
+    _ensureObject(options, value) {
+        let name = options;
+        if(typeof name === "string") {
+            options = {};
+            options[name] = value;
+        }
+
+        return options;
+    }
+
+    _getValue(options, name, unwrapObservables) {
+        let getter = this.cachedGetters[name];
+        if(!getter) {
+            getter = this.cachedGetters[name] = coreDataUtils.compileGetter(name);
+        }
+
+        return getter(options, { functionsAsIs: true, unwrapObservables });
+    }
+
+    _setSilent(options, rulesOptions) {
+        extend(true, options, rulesOptions);
+
+        for(const fieldName in this._optionsByReference) {
+            if(Object.prototype.hasOwnProperty.call(rulesOptions, fieldName)) {
+                options[fieldName] = rulesOptions[fieldName];
+            }
+        }
+    }
+
+    onChanging(callBack) {
+        this._changingCallbacks.add(callBack);
+    }
+
+    onChanged(callBack) {
+        this._changedCallbacks.add(callBack);
+    }
+
+    onDeprecated(callBack) {
+        this._deprecatedCallbacks.add(callBack);
+    }
+
+    getValue(name) {
+        const normalizedName = this._normalizeName(name);
+        return this._getValue(this._options, normalizedName);
```

Comment:
No need to extract a variable
```suggestion
        return this._getValue(this._options, this._normalizeName(name));
```